src/main/java/org/broadinstitute/hellbender/tools/AnnotatePairOrientation.java

```diff
@@ -0,0 +1,119 @@
+                "\n  The output of this tool will only count reads that fully overlap (and match) the variant or reference sequence (this is relevant for indels)." +
+                "\n  IMPORTANT:  This tool cannot produce the exact same F1R2/F2R1 counts as you would get from M2, due to the nature of how M2 calls variants (using read likelihoods, wheras this tool uses a base quality filter).",
+        oneLineSummary = "(EXPERIMENTAL) Annotate a non-M2 VCF (using the associated tumor bam) with pair orientation fields (e.g. " + GATKVCFConstants.F1R2_KEY + " ).",
+        programGroup = VariantProgramGroup.class
+)
+@BetaFeature
+public class AnnotatePairOrientation extends VariantWalker {
+
+    @Argument(
+            doc = "Output Somatic SNP/Indel VCF file with additional annotations.",
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME)
+    protected File outputFile;
+
+    public final static String CUTOFF_SHORT_NAME = "c";
+    public final static String CUTOFF_LONG_NAME = "meanBaseQualityCutoff";
+    public final static double MEAN_BASE_QUALITY_DEFAULT_CUTOFF = 7.0;
+    @Argument(
+            doc = "Cutoff for the mean base quality value(s) to count the read.  These are for bases that overlap the variant.",
+            shortName = CUTOFF_SHORT_NAME, fullName = CUTOFF_LONG_NAME, minValue = 0.0, maxRecommendedValue = 10.0,
+            optional = true
+    )
+    private Double meanBaseQualityCutoff = MEAN_BASE_QUALITY_DEFAULT_CUTOFF;
+
+    private VariantContextWriter vcfWriter;
+
+    @Override
+    public void onTraversalStart() {
+        vcfWriter = createVCFWriter(outputFile);
+        vcfWriter.writeHeader(createVCFHeader(getHeaderForVariants(), getCommandLine()));
+    }
+
+    @Override
+    public List<ReadFilter> getDefaultReadFilters() {
+        return Mutect2Engine.makeStandardMutect2ReadFilters();
+    }
+
+    @Override
+    public void apply(VariantContext variant, ReadsContext readsContext, ReferenceContext referenceContext, FeatureContext featureContext) {
+
+
+        final ReadPileup readPileup = GATKProtectedVariantContextUtils.getPileup(variant, readsContext);
+        final List<Genotype> updatedGenotypes = new ArrayList<>();
+
+        final Map<String, ReadPileup> sampleToReadPileup = readPileup.splitBySample(getHeaderForReads(), null);
+
+        for (Genotype g : variant.getGenotypes()) {
+
+            final ReadPileup genotypeSamplePileup = sampleToReadPileup.get(g.getSampleName());
+            final GenotypeBuilder gb = new GenotypeBuilder(g);
+            OxoGReadCounts.annotateSingleVariant(variant, gb, genotypeSamplePileup, meanBaseQualityCutoff);
+            updatedGenotypes.add(gb.make());
+        }
+
+        final VariantContextBuilder vcb = new VariantContextBuilder(variant);
+        vcb.genotypes(updatedGenotypes);
+        vcfWriter.add(vcb.make());
```

Comment:
It's more idiomatic IMO to exploit the Builder pattern with `vcfWriter.add(new VariantContextBuilder(variant).genotypes(updatedGenotypes).make())`

Reply:
Done